src/FffGcodeWriter.cpp

```diff
@@ -712,15 +712,39 @@ LayerPlan& FffGcodeWriter::processLayer(const SliceDataStorage& storage, int lay
     return gcode_layer;
 }
 
-bool FffGcodeWriter::getExtrudersNeedPrimeDuringFirstLayer() const
+bool FffGcodeWriter::getAnyExtruderNeedPrimeDuringFirstLayer(const SliceDataStorage& storage) const
 {
-    switch(gcode.getFlavor())
+    bool need_prime = false;
+    switch (gcode.getFlavor())
     {
         case EGCodeFlavor::GRIFFIN:
-            return true;
+            need_prime = true;
+            break;
         default:
-            return false; // TODO: change this once priming for other firmware types is implemented
+            need_prime = false; // TODO: change this once priming for other firmware types is implemented
+            break;
     }
+
+    // check the settings if the prime poop is disabled
+    if (need_prime)
+    {
+        bool has_extruder_that_needs_prime = false;
+        const std::vector<bool> extruders_is_used_overall = storage.getExtrudersUsed();
+
+        for (uint32_t extruder_nr = 0; extruder_nr < extruders_is_used_overall.size(); ++extruder_nr)
+        {
+            const bool extruder_prime_enabled = storage.getExtruderPrimeEnabled(extruder_nr);
```

Comment:
This variable name is not correct. It should be extruder_prime_poop_enabled

The word poop is important here. An extruder should always be primed before it starts.
Such priming may include extruder start gcode, the G280 gcode or it may even be empty.